packages/x-charts-pro/src/context/ZoomProvider/ZoomProvider.tsx

```diff
@@ -34,33 +21,109 @@ export function ZoomProvider({ children, xAxis, yAxis }: ZoomProviderProps) {
     [xAxis, yAxis],
   );
 
-  const [zoomData, setZoomData] = React.useState<ZoomData[]>(() =>
-    Object.values(options).map(({ axisId, minStart: start, maxEnd: end }) => ({
-      axisId,
-      start,
-      end,
-    })),
+  const [zoomData, setZoomData] = useControlled<ZoomData[]>({
+    controlled: zoom,
+    default: [],
+    name: 'ZoomProvider',
+    state: 'zoom',
+  });
+
+  const setZoomDataCallback = React.useCallback(
+    (newZoomData: Parameters<ZoomState['setZoomData']>[0]) => {
+      setZoomData(newZoomData);
+      onZoomChange?.(newZoomData);
+    },
+    [setZoomData, onZoomChange],
   );
 
   const value = React.useMemo(
-    () => ({
-      isInitialized: true,
-      data: {
-        isZoomEnabled: zoomData.length > 0,
-        isPanEnabled: isPanEnabled(options),
-        options,
-        zoomData,
-        setZoomData,
-        isInteracting,
-        setIsInteracting,
-      },
-    }),
-    [zoomData, setZoomData, isInteracting, setIsInteracting, options],
+    () =>
+      ({
+        isInitialized: true,
+        data: {
+          isZoomEnabled: Object.keys(options).length > 0,
+          isPanEnabled: isPanEnabled(options),
+          options,
+          zoomData,
+          setZoomData: setZoomDataCallback,
+          isInteracting,
+          setIsInteracting,
+        },
+      }) satisfies Initializable<ZoomState>,
```

Comment:
@LukasTy last time we tried to use `satisfies` we had issues with the build, do you know it that's still the case?

Reply:
Moved the typing to `useMemo<T>` instead

Reply:
Clarification—satisfies still fails `docs:api` until we upgrade or drop `react-docgen`. 😉